In [ ]:
install.packages("NeuroDataSets")

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [ ]:
library(NeuroDataSets)

In [ ]:
data(ASD_risks_tbl_df)

In [ ]:
ASD_risks_tbl_df

CASE_NO_PATIENT'S,A1,A2,A3,A4,A5,A6,A7,A8,A9,⋯,Global developmental delay/intellectual disability,Social/Behavioural Issues,Childhood Autism Rating Scale,Anxiety_disorder,Sex,Ethnicity,Jaundice,Family_mem_with_ASD,Who_completed_the_test,ASD_traits
<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,⋯,<chr>,<chr>,<dbl>,<chr>,<chr>,<chr>,<chr>,<chr>,<chr>,<chr>
1,0,0,0,0,0,0,1,1,0,⋯,Yes,Yes,1,Yes,F,middle eastern,Yes,No,Family Member,No
2,1,1,0,0,0,1,1,0,0,⋯,Yes,Yes,2,Yes,M,White European,Yes,No,Family Member,Yes
3,1,0,0,0,0,0,1,1,0,⋯,Yes,Yes,4,Yes,M,Middle Eastern,Yes,No,Family Member,Yes
4,1,1,1,1,1,1,1,1,1,⋯,Yes,Yes,2,Yes,M,Hispanic,No,No,Family Member,Yes
5,1,1,0,1,1,1,1,1,1,⋯,Yes,Yes,1,Yes,F,White European,No,No,Family Member,Yes
6,1,1,0,0,1,1,1,1,1,⋯,Yes,Yes,2,Yes,M,Black,No,No,Family Member,Yes
7,1,0,0,1,1,1,0,0,1,⋯,Yes,Yes,4,Yes,M,Asian,Yes,No,Family Member,Yes
8,0,1,0,0,1,0,1,1,1,⋯,Yes,Yes,2,Yes,M,Asian,Yes,No,Family Member,Yes
9,0,0,0,0,0,0,1,0,0,⋯,Yes,Yes,2,Yes,M,Asian,No,No,Family Member,No


In [ ]:
asd <- ASD_risks_tbl_df

In [ ]:
install.packages("janitor")

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

also installing the dependency ‘snakecase’




In [ ]:
library(janitor)


Attaching package: ‘janitor’


The following objects are masked from ‘package:stats’:

    chisq.test, fisher.test




In [ ]:
asd_clean <- clean_names(asd)    # Add consistency in variable or column names
asd_clean

case_no_patients,a1,a2,a3,a4,a5,a6,a7,a8,a9,⋯,global_developmental_delay_intellectual_disability,social_behavioural_issues,childhood_autism_rating_scale,anxiety_disorder,sex,ethnicity,jaundice,family_mem_with_asd,who_completed_the_test,asd_traits
<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,⋯,<chr>,<chr>,<dbl>,<chr>,<chr>,<chr>,<chr>,<chr>,<chr>,<chr>
1,0,0,0,0,0,0,1,1,0,⋯,Yes,Yes,1,Yes,F,middle eastern,Yes,No,Family Member,No
2,1,1,0,0,0,1,1,0,0,⋯,Yes,Yes,2,Yes,M,White European,Yes,No,Family Member,Yes
3,1,0,0,0,0,0,1,1,0,⋯,Yes,Yes,4,Yes,M,Middle Eastern,Yes,No,Family Member,Yes
4,1,1,1,1,1,1,1,1,1,⋯,Yes,Yes,2,Yes,M,Hispanic,No,No,Family Member,Yes
5,1,1,0,1,1,1,1,1,1,⋯,Yes,Yes,1,Yes,F,White European,No,No,Family Member,Yes
6,1,1,0,0,1,1,1,1,1,⋯,Yes,Yes,2,Yes,M,Black,No,No,Family Member,Yes
7,1,0,0,1,1,1,0,0,1,⋯,Yes,Yes,4,Yes,M,Asian,Yes,No,Family Member,Yes
8,0,1,0,0,1,0,1,1,1,⋯,Yes,Yes,2,Yes,M,Asian,Yes,No,Family Member,Yes
9,0,0,0,0,0,0,1,0,0,⋯,Yes,Yes,2,Yes,M,Asian,No,No,Family Member,No


In [ ]:
install.packages("mice")   # only once
library(mice)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

also installing the dependencies ‘rbibutils’, ‘Rdpack’, ‘minqa’, ‘nloptr’, ‘reformulas’, ‘ucminf’, ‘numDeriv’, ‘iterators’, ‘lme4’, ‘ordinal’, ‘foreach’, ‘shape’, ‘RcppEigen’, ‘pan’, ‘jomo’, ‘glmnet’, ‘mitml’



Attaching package: ‘mice’


The following object is masked from ‘package:stats’:

    filter


The following objects are masked from ‘package:base’:

    cbind, rbind




In [ ]:
install.packages("dplyr")

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [ ]:
library(dplyr)


Attaching package: ‘dplyr’


The following objects are masked from ‘package:stats’:

    filter, lag


The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union




In [ ]:
install.packages("stringr")

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [ ]:
library(stringr)

In [ ]:
asd_mice_0 <- asd_clean |>
  # 1. Lowercase, trim whitespace, and Title-Case all character variables to standardize categories under categorical columns
  mutate(across(
    where(is.character),
    ~ str_to_title(str_to_lower(str_trim(.x)))
  ))

In [ ]:
asd_mice <- asd_mice_0 %>%
  mutate(
    across(where(is.character), as.factor),
    across(
      c(a1:a9, a10_autism_spectrum_quotient),
      ~ factor(.x, levels = c(0, 1))
    )
  )

  asd_mice

case_no_patients,a1,a2,a3,a4,a5,a6,a7,a8,a9,⋯,global_developmental_delay_intellectual_disability,social_behavioural_issues,childhood_autism_rating_scale,anxiety_disorder,sex,ethnicity,jaundice,family_mem_with_asd,who_completed_the_test,asd_traits
<dbl>,<fct>,<fct>,<fct>,<fct>,<fct>,<fct>,<fct>,<fct>,<fct>,⋯,<fct>,<fct>,<dbl>,<fct>,<fct>,<fct>,<fct>,<fct>,<fct>,<fct>
1,0,0,0,0,0,0,1,1,0,⋯,Yes,Yes,1,Yes,F,Middle Eastern,Yes,No,Family Member,No
2,1,1,0,0,0,1,1,0,0,⋯,Yes,Yes,2,Yes,M,White European,Yes,No,Family Member,Yes
3,1,0,0,0,0,0,1,1,0,⋯,Yes,Yes,4,Yes,M,Middle Eastern,Yes,No,Family Member,Yes
4,1,1,1,1,1,1,1,1,1,⋯,Yes,Yes,2,Yes,M,Hispanic,No,No,Family Member,Yes
5,1,1,0,1,1,1,1,1,1,⋯,Yes,Yes,1,Yes,F,White European,No,No,Family Member,Yes
6,1,1,0,0,1,1,1,1,1,⋯,Yes,Yes,2,Yes,M,Black,No,No,Family Member,Yes
7,1,0,0,1,1,1,0,0,1,⋯,Yes,Yes,4,Yes,M,Asian,Yes,No,Family Member,Yes
8,0,1,0,0,1,0,1,1,1,⋯,Yes,Yes,2,Yes,M,Asian,Yes,No,Family Member,Yes
9,0,0,0,0,0,0,1,0,0,⋯,Yes,Yes,2,Yes,M,Asian,No,No,Family Member,No


In [ ]:
lapply(asd_mice[sapply(asd_mice, is.factor)], table, useNA = "ifany")

$a1

   0    1 
1391  594 

$a2

   0    1 
1512  473 

$a3

   0    1 
1562  423 

$a4

   0    1 
1445  540 

$a5

   0    1 
1432  553 

$a6

   0    1 
1377  608 

$a7

   0    1 
1300  685 

$a8

   0    1 
1501  484 

$a9

   0    1 
1469  516 

$a10_autism_spectrum_quotient

   0    1 
1099  886 

$speech_delay_language_disorder

  No  Yes 
 928 1057 

$learning_disorder

  No  Yes 
 910 1075 

$genetic_disorders

  No  Yes 
 972 1013 

$depression

  No  Yes <NA> 
 930 1054    1 

$global_developmental_delay_intellectual_disability

  No  Yes 
 931 1054 

$social_behavioural_issues

  No  Yes <NA> 
 931 1040   14 

$anxiety_disorder

  No  Yes 
 934 1051 

$sex

   F    M 
 538 1447 

$ethnicity

         Asian          Black       Hispanic         Latino Middle Eastern 
           605             53             40             26            403 
         Mixed  Native Indian         Others       Pacifica    South Asian 
             8              3             35              

Standardized categories names obtained.

In [ ]:
str(asd_mice)

spc_tbl_ [1,985 × 28] (S3: spec_tbl_df/tbl_df/tbl/data.frame)
 $ case_no_patients                                  : num [1:1985] 1 2 3 4 5 6 7 8 9 10 ...
 $ a1                                                : Factor w/ 2 levels "0","1": 1 2 2 2 2 2 2 1 1 2 ...
 $ a2                                                : Factor w/ 2 levels "0","1": 1 2 1 2 2 2 1 2 1 2 ...
 $ a3                                                : Factor w/ 2 levels "0","1": 1 1 1 2 1 1 1 1 1 2 ...
 $ a4                                                : Factor w/ 2 levels "0","1": 1 1 1 2 2 1 2 1 1 1 ...
 $ a5                                                : Factor w/ 2 levels "0","1": 1 1 1 2 2 2 2 2 1 2 ...
 $ a6                                                : Factor w/ 2 levels "0","1": 1 2 1 2 2 2 2 1 1 2 ...
 $ a7                                                : Factor w/ 2 levels "0","1": 2 2 2 2 2 2 1 2 2 1 ...
 $ a8                                                : Factor w/ 2 levels "0","1": 2 1 2 2 2 2 1

In [ ]:
ini <- mice(asd_mice, maxit = 0)

ini$method

Warning message:
“Number of logged events: 1”


case_no_patients 
                                                "" 
                                                a1 
                                                "" 
                                                a2 
                                                "" 
                                                a3 
                                                "" 
                                                a4 
                                                "" 
                                                a5 
                                                "" 
                                                a6 
                                                "" 
                                                a7 
                                                "" 
                                                a8 
                                                "" 
                                                a9 
                                                "" 
                      a10_autism_spectrum_quotient 
                                                "" 
                       social_responsiveness_scale 
                                             "pmm" 
                                         age_years 
                                                "" 
                                    qchat_10_score 
                                             "pmm" 
                    speech_delay_language_disorder 
                                                "" 
                                 learning_disorder 
                                                "" 
                                 genetic_disorders 
                                                "" 
                                        depression 
                                          "logreg" 
global_developmental_delay_intellectual_disability 
                                                "" 
                         social_behavioural_issues 
                                                "" 
                     childhood_autism_rating_scale 
                                                "" 
                                  anxiety_disorder 
                                                "" 
                                               sex 
                                                "" 
                                         ethnicity 
                                                "" 
                                          jaundice 
                                                "" 
                               family_mem_with_asd 
                                                "" 
                            who_completed_the_test 
                                                "" 
                                        asd_traits 
                                                ""

social_behavioural_issues, even though missing are not assigned a method.

In [ ]:
class(asd_mice$social_behavioural_issues)

table(asd_mice$social_behavioural_issues, useNA = "ifany")

[1] "factor"


  No  Yes <NA> 
 931 1040   14 

It is indeed a factor with missing variables.

In [ ]:
ini$loggedEvents

,it,im,dep,meth,out
,<dbl>,<dbl>,<chr>,<chr>,<chr>
1,0,0,,collinear,social_behavioural_issues


It turns out to be collinear.

In [ ]:
pred <- ini$predictorMatrix

pred["social_behavioural_issues", ]

case_no_patients 
                                                 0 
                                                a1 
                                                 0 
                                                a2 
                                                 0 
                                                a3 
                                                 0 
                                                a4 
                                                 0 
                                                a5 
                                                 0 
                                                a6 
                                                 0 
                                                a7 
                                                 0 
                                                a8 
                                                 0 
                                                a9 
                                                 0 
                      a10_autism_spectrum_quotient 
                                                 0 
                       social_responsiveness_scale 
                                                 0 
                                         age_years 
                                                 0 
                                    qchat_10_score 
                                                 0 
                    speech_delay_language_disorder 
                                                 0 
                                 learning_disorder 
                                                 0 
                                 genetic_disorders 
                                                 0 
                                        depression 
                                                 0 
global_developmental_delay_intellectual_disability 
                                                 0 
                         social_behavioural_issues 
                                                 0 
                     childhood_autism_rating_scale 
                                                 0 
                                  anxiety_disorder 
                                                 0 
                                               sex 
                                                 0 
                                         ethnicity 
                                                 0 
                                          jaundice 
                                                 0 
                               family_mem_with_asd 
                                                 0 
                            who_completed_the_test 
                                                 0 
                                        asd_traits 
                                                 0

For all other variables, social_behavioral_issues are dropped.

In [ ]:
pred

,case_no_patients,a1,a2,a3,a4,a5,a6,a7,a8,a9,⋯,global_developmental_delay_intellectual_disability,social_behavioural_issues,childhood_autism_rating_scale,anxiety_disorder,sex,ethnicity,jaundice,family_mem_with_asd,who_completed_the_test,asd_traits
case_no_patients,0,1,1,1,1,1,1,1,1,1,⋯,1,0,1,1,1,1,1,1,1,1
a1,1,0,1,1,1,1,1,1,1,1,⋯,1,0,1,1,1,1,1,1,1,1
a2,1,1,0,1,1,1,1,1,1,1,⋯,1,0,1,1,1,1,1,1,1,1
a3,1,1,1,0,1,1,1,1,1,1,⋯,1,0,1,1,1,1,1,1,1,1
a4,1,1,1,1,0,1,1,1,1,1,⋯,1,0,1,1,1,1,1,1,1,1
a5,1,1,1,1,1,0,1,1,1,1,⋯,1,0,1,1,1,1,1,1,1,1
a6,1,1,1,1,1,1,0,1,1,1,⋯,1,0,1,1,1,1,1,1,1,1
a7,1,1,1,1,1,1,1,0,1,1,⋯,1,0,1,1,1,1,1,1,1,1
a8,1,1,1,1,1,1,1,1,0,1,⋯,1,0,1,1,1,1,1,1,1,1
a9,1,1,1,1,1,1,1,1,1,0,⋯,1,0,1,1,1,1,1,1,1,1


In [ ]:
pred[c(
  "qchat_10_score",
  "social_responsiveness_scale",
  "depression",
  "social_behavioural_issues"
), ]

,case_no_patients,a1,a2,a3,a4,a5,a6,a7,a8,a9,⋯,global_developmental_delay_intellectual_disability,social_behavioural_issues,childhood_autism_rating_scale,anxiety_disorder,sex,ethnicity,jaundice,family_mem_with_asd,who_completed_the_test,asd_traits
qchat_10_score,1,1,1,1,1,1,1,1,1,1,⋯,1,0,1,1,1,1,1,1,1,1
social_responsiveness_scale,1,1,1,1,1,1,1,1,1,1,⋯,1,0,1,1,1,1,1,1,1,1
depression,1,1,1,1,1,1,1,1,1,1,⋯,1,0,1,1,1,1,1,1,1,1
social_behavioural_issues,0,0,0,0,0,0,0,0,0,0,⋯,0,0,0,0,0,0,0,0,0,0


In [ ]:
pred0 <- make.predictorMatrix(asd_mice)

pred0["social_behavioural_issues", ]

case_no_patients 
                                                 1 
                                                a1 
                                                 1 
                                                a2 
                                                 1 
                                                a3 
                                                 1 
                                                a4 
                                                 1 
                                                a5 
                                                 1 
                                                a6 
                                                 1 
                                                a7 
                                                 1 
                                                a8 
                                                 1 
                                                a9 
                                                 1 
                      a10_autism_spectrum_quotient 
                                                 1 
                       social_responsiveness_scale 
                                                 1 
                                         age_years 
                                                 1 
                                    qchat_10_score 
                                                 1 
                    speech_delay_language_disorder 
                                                 1 
                                 learning_disorder 
                                                 1 
                                 genetic_disorders 
                                                 1 
                                        depression 
                                                 1 
global_developmental_delay_intellectual_disability 
                                                 1 
                         social_behavioural_issues 
                                                 0 
                     childhood_autism_rating_scale 
                                                 1 
                                  anxiety_disorder 
                                                 1 
                                               sex 
                                                 1 
                                         ethnicity 
                                                 1 
                                          jaundice 
                                                 1 
                               family_mem_with_asd 
                                                 1 
                            who_completed_the_test 
                                                 1 
                                        asd_traits 
                                                 1

In the initial matrix, it was not dropped for other variables but after collinearity was found, it was.

In [ ]:
vars_to_check <- c(
  "speech_delay_language_disorder",
  "learning_disorder",
  "genetic_disorders",
  "depression",
  "global_developmental_delay_intellectual_disability",
  "anxiety_disorder",
  "sex",
  "jaundice",
  "family_mem_with_asd",
  "asd_traits"
)

lapply(
  vars_to_check,
  function(v) {
    list(
      variable = v,
      table = table(
        asd_mice$social_behavioural_issues,
        asd_mice[[v]],
        useNA = "ifany"
      )
    )
  }
)

[[1]]
[[1]]$variable
[1] "speech_delay_language_disorder"

[[1]]$table
      
         No  Yes
  No    928    3
  Yes     0 1040
  <NA>    0   14


[[2]]
[[2]]$variable
[1] "learning_disorder"

[[2]]$table
      
         No  Yes
  No    910   21
  Yes     0 1040
  <NA>    0   14


[[3]]
[[3]]$variable
[1] "genetic_disorders"

[[3]]$table
      
        No Yes
  No   931   0
  Yes   41 999
  <NA>   0  14


[[4]]
[[4]]$variable
[1] "depression"

[[4]]$table
      
         No  Yes <NA>
  No    930    0    1
  Yes     0 1040    0
  <NA>    0   14    0


[[5]]
[[5]]$variable
[1] "global_developmental_delay_intellectual_disability"

[[5]]$table
      
         No  Yes
  No    930    1
  Yes     1 1039
  <NA>    0   14


[[6]]
[[6]]$variable
[1] "anxiety_disorder"

[[6]]$table
      
         No  Yes
  No    931    0
  Yes     3 1037
  <NA>    0   14


[[7]]
[[7]]$variable
[1] "sex"

[[7]]$table
      
         F   M
  No   370 561
  Yes  165 875
  <NA>   3  11


[[8]]
[[8]]$variable
[1] "j

Every single patient with social_behavioural_issues == Yes has depression == Yes, and every patient with social_behavioural_issues == No has depression == No. Because these two columns contain identical information across all 1,970 complete cases, mice detected linear dependency and automatically dropped social_behavioural_issues.

speech_delay_language_disorder & learning_disorder: Whenever social_behavioural_issues == Yes, 100% of patients (1,040 / 1,040) also have Yes for these conditions. There are zero No cases.

Hence, social_behavioral_issues were dropped.

Let's check whether there is any possibility of MAR in social_behavioral issues.

In [ ]:
table(
  Respondent = asd_mice$who_completed_the_test,
  Missing_Social_Issues = is.na(asd_mice$social_behavioural_issues),
  useNA = "ifany"
)  # Checking for structural missingness; If all missing values are under one category, it means that that variable can explain
   # can explain the missigness. Hence, MAR possibility.

                          Missing_Social_Issues
Respondent                 FALSE TRUE
  Family Member              716    0
  Health Care Professional  1219   14
  Others                       3    0
  School And Ngo              29    0
  Self                         4    0

All missing Social behavioral issues are explained by the respondent group - Health Care Professional. So, possibility of MAR.

In [ ]:
# 2. Age Distribution across Missing vs. Observed groups
asd_mice |>
  group_by(Missing_Social_Issues = is.na(social_behavioural_issues)) |>
  summarise(
    Count = n(),
    Min_Age = min(age_years, na.rm = TRUE),
    Median_Age = median(age_years, na.rm = TRUE),
    Max_Age = max(age_years, na.rm = TRUE),
    Mean_Age = round(mean(age_years, na.rm = TRUE), 2)
  ) # Checking for structural missingness; If all missing values are under age group, it means that that variable can explain
   # can explain the missigness. Hence, MAR possibility.

Missing_Social_Issues,Count,Min_Age,Median_Age,Max_Age,Mean_Age
<lgl>,<int>,<dbl>,<dbl>,<dbl>,<dbl>
FALSE,1971,1,9.0,18,9.62
TRUE,14,3,8.5,17,9.79


Age distribution is same across both groups. Could be MCAR. Overall, till now, the results point to MAR wrt respondent & social behavioral issues and asd_traits & qchat10 at least. So, there is a possibility that it is not completely MNAR. Even the collinear variable in MICE shows the possiblity of MAR. So, relying on only complete cases might not be the best even if some other variables show the possibility of MCAR. MICE implementation was justified.

However, with social_behavioral_issues being highly explained by depression and also other variables, it would be wise to remove it.

And the different autism scores are meant for different age groups.

AQ10 is meant only for 16 and older.
CARS is meant for 2 and above.
SRS is meant for 4-18 year olds.
QChat10 is meant for 18-24 month olds.
So, the scores are not standardized.

Also, they could be highly correlated to ASD traits as they are used for the screening itself. So, dropping them would be better.

In [ ]:
asd_model <- asd_mice %>%
  select(
    -a1,
    -a2,
    -a3,
    -a4,
    -a5,
    -a6,
    -a7,
    -a8,
    -a9,
    -a10_autism_spectrum_quotient,
    -social_responsiveness_scale,
    -qchat_10_score,
    -childhood_autism_rating_scale,
    -social_behavioural_issues
  )

  asd_model

case_no_patients,age_years,speech_delay_language_disorder,learning_disorder,genetic_disorders,depression,global_developmental_delay_intellectual_disability,anxiety_disorder,sex,ethnicity,jaundice,family_mem_with_asd,who_completed_the_test,asd_traits
<dbl>,<dbl>,<fct>,<fct>,<fct>,<fct>,<fct>,<fct>,<fct>,<fct>,<fct>,<fct>,<fct>,<fct>
1,2,Yes,Yes,Yes,Yes,Yes,Yes,F,Middle Eastern,Yes,No,Family Member,No
2,3,Yes,Yes,Yes,Yes,Yes,Yes,M,White European,Yes,No,Family Member,Yes
3,3,Yes,Yes,Yes,Yes,Yes,Yes,M,Middle Eastern,Yes,No,Family Member,Yes
4,2,Yes,Yes,No,Yes,Yes,Yes,M,Hispanic,No,No,Family Member,Yes
5,2,Yes,Yes,No,Yes,Yes,Yes,F,White European,No,No,Family Member,Yes
6,2,Yes,Yes,No,Yes,Yes,Yes,M,Black,No,No,Family Member,Yes
7,3,Yes,Yes,Yes,Yes,Yes,Yes,M,Asian,Yes,No,Family Member,Yes
8,3,Yes,Yes,Yes,Yes,Yes,Yes,M,Asian,Yes,No,Family Member,Yes
9,3,Yes,Yes,No,Yes,Yes,Yes,M,Asian,No,No,Family Member,No


In [ ]:
dim(asd_model)

[1] 1985   14

In [ ]:
colSums(is.na(asd_model))

case_no_patients 
                                                 0 
                                         age_years 
                                                 0 
                    speech_delay_language_disorder 
                                                 0 
                                 learning_disorder 
                                                 0 
                                 genetic_disorders 
                                                 0 
                                        depression 
                                                 1 
global_developmental_delay_intellectual_disability 
                                                 0 
                                  anxiety_disorder 
                                                 0 
                                               sex 
                                                 0 
                                         ethnicity 
                                                 0 
                                          jaundice 
                                                 0 
                               family_mem_with_asd 
                                                 0 
                            who_completed_the_test 
                                                 0 
                                        asd_traits 
                                                 0

Only one row has missing data with depression, so we drop it.

In [ ]:
asd_model_complete <- asd_model %>%
  tidyr::drop_na()

asd_model_complete

case_no_patients,age_years,speech_delay_language_disorder,learning_disorder,genetic_disorders,depression,global_developmental_delay_intellectual_disability,anxiety_disorder,sex,ethnicity,jaundice,family_mem_with_asd,who_completed_the_test,asd_traits
<dbl>,<dbl>,<fct>,<fct>,<fct>,<fct>,<fct>,<fct>,<fct>,<fct>,<fct>,<fct>,<fct>,<fct>
1,2,Yes,Yes,Yes,Yes,Yes,Yes,F,Middle Eastern,Yes,No,Family Member,No
2,3,Yes,Yes,Yes,Yes,Yes,Yes,M,White European,Yes,No,Family Member,Yes
3,3,Yes,Yes,Yes,Yes,Yes,Yes,M,Middle Eastern,Yes,No,Family Member,Yes
4,2,Yes,Yes,No,Yes,Yes,Yes,M,Hispanic,No,No,Family Member,Yes
5,2,Yes,Yes,No,Yes,Yes,Yes,F,White European,No,No,Family Member,Yes
6,2,Yes,Yes,No,Yes,Yes,Yes,M,Black,No,No,Family Member,Yes
7,3,Yes,Yes,Yes,Yes,Yes,Yes,M,Asian,Yes,No,Family Member,Yes
8,3,Yes,Yes,Yes,Yes,Yes,Yes,M,Asian,Yes,No,Family Member,Yes
9,3,Yes,Yes,No,Yes,Yes,Yes,M,Asian,No,No,Family Member,No


In [ ]:
dim(asd_model_complete)

[1] 1984   14

In [ ]:
write.csv(asd_model_complete, "asd_model.csv", row.names = FALSE)